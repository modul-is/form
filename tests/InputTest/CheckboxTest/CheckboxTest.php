<?php

declare(strict_types=1);

namespace ModulIS\Form\Tests;

require_once __DIR__ . '/../../bootstrap.php';

use Tester\Assert;

class CheckboxTest extends TestCase
{
	public function testRender()
	{
		$form = $this->getForm();

		$form->addCheckbox('check', 'Check');

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/basic.latte'));

		Assert::same($html, $form->getComponent('check')->render()->__toString());
	}


	public function testRenderOptionId()
	{
		$form = $this->getForm();

		$form->addCheckbox('check', 'Check')
			->setOption('id', 'customId');

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/id.latte'));

		Assert::same($html, $form->getComponent('check')->render()->__toString());
	}


	public function testRenderCustomTemplate()
	{
		$form = $this->getForm();

		$form->addCheckbox('check', 'Check')
			->setTemplate(__DIR__ . '/custom.latte');

		$checkboxString = 'custom-template';

		Assert::same($checkboxString, $form->getComponent('check')->render());
	}


	public function testRenderHidden()
	{
		$form = $this->getForm();

		$form->addCheckbox('check', 'Check')
			->setOption('hide', true);

		$checkboxString = '';

		Assert::same($checkboxString, $form->getComponent('check')->render());
	}


	public function testRenderSkip()
	{
		$form = $this->getForm();

		$form->addCheckbox('check', 'Check')
			->setAutoRenderSkip();

		$checkboxString = '';

		Assert::same($checkboxString, $form->getComponent('check')->render());
	}


	public function testRenderTooltip()
	{
		$form = $this->getForm();

		$form->addCheckbox('check', 'Check')
			->setTooltip('MyTooltip');

		$html = str_replace(["\t", "\n", "\r"], '', file_get_contents(__DIR__ . '/tooltip.latte'));

		Assert::same($html, $form->getComponent('check')->render()->__toString());
	}
}

$testcase = new CheckboxTest;
$testcase->run();
