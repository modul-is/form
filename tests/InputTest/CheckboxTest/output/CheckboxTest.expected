<div class="mb-3 row"><div class="col-sm-8 offset-sm-4"><div class="form-check"><input type="checkbox" name="check" id="frm-form-check" class="form-check-input "><label for="frm-form-check" class="form-check-label">Check</label><span title="MyTooltip" data-placement="top" data-togle="tooltip"><span class="fal fa-question-circle color-blue fa-fw"></span></span></div></div></div>